fury/stream/widget.py

```diff
@@ -0,0 +1,254 @@
+import subprocess
+import numpy as np
+import sys
+if sys.version_info.minor >= 8:
+    PY_VERSION_8 = True
+else:
+    PY_VERSION_8 = False
+try:
+    from IPython.display import IFrame
+    from IPython.core.display import display
+    IPYTHON_AVAILABLE = True
+except ImportError:
+    IPYTHON_AVAILABLE = False
+
+import time
+import socket
+import errno
+
+from fury.stream.client import FuryStreamClient, FuryStreamInteraction
+
+
+def check_port_is_available(host, port):
+    """Check if a given port it's available
+
+    Parameters
+    ----------
+    host : str
+    port : int
+
+    Returns
+    -------
+    available : bool
+
+    """
+    available = True
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        s.bind((host, port))
+    except socket.error as e:
+        if e.errno == errno.EADDRINUSE:
+            available = False
+    s.close()
+    return available
+
+
+class Widget:
+    def __init__(
+            self, showm, ms_stream=33,
+            ms_interaction=33,
+            host='localhost', port=None,
+            encoding='mjpeg', ms_jpeg=33,
+            queue_size=20):
+        """Thi Obj it's able execute the fury streaming system
```

Comment:
Please add more information about this.

Reply:
About what this class actually does.

Reply:
It doesn't only execute the fury streaming system. Correct?

Reply:
No, this also creates a process to start the webrtc enconding.

Reply:
This object is able to execute the FURY streaming system ...